kernel/e1000/src/regs.rs

```diff
@@ -1,3 +1,169 @@
+//! This file contains the structs that are used to access device registers.
+//! The registers are divided into multiple structs because we need to separate out the 
+//! receive and transmit queue registers and store them separately in a per-queue struct. 
+//! Though the e1000 device only has 1 pair of receive and transmit queues, we still structure
+//! the design this way to be able to use code shared by all network drivers.
+//! 
+//! The 4 structs which cover the registers of the entire memory-mapped region are:
+//! * `E1000Registers`
+//! * `E1000RxRegisters`
+//! * `E1000TxRegisters`
+//! * `E1000MacRegisters`
```

Comment:
Great docs! 

I'd recommend having a one-line summary at the top of every file, since that's what Rustdoc expects for its "short one-liner" brief of a module or crate.